tools/tasks.py

```diff
@@ -479,15 +478,15 @@ def add_gzip_file_task(sourcefile_path, salt, content_postprocessors=[]):
     """
     def python_add_file_task():
         with gzip.open(sourcefile_path, 'r') as infile:
-            contents = infile.read()
+            contents = infile.read().decode('utf-8')
             for content_postprocessor in content_postprocessors:
                 contents = content_postprocessor(contents)
             redactor = LogRedactor(salt, tempfile.mkdtemp())
             contents = redactor.redact_string(contents)
 
-        out = StringIO.StringIO()
+        out = io.BytesIO()
         with gzip.GzipFile(fileobj=out, mode="w") as f:
-            f.write(contents)
+            f.write(contents.encode())
```

Comment:
Elsewhere you're doing try/catch for errors on encode() - should that be happening here?

Reply:
No. We know the contents came from file and know we've converted them to string and need encoding back.